include/knowhere/index/index_factory.h

```diff
@@ -29,10 +29,15 @@ class IndexFactory {
     Create(const std::string& name, const int32_t& version, const Object& object = nullptr);
     template <typename DataType>
     const IndexFactory&
-    Register(const std::string& name, std::function<Index<IndexNode>(const int32_t&, const Object&)> func);
+    Register(const std::string& name, std::function<Index<IndexNode>(const int32_t&, const Object&)> func,
+             const uint64_t features);
     static IndexFactory&
     Instance();
     typedef std::tuple<std::set<std::pair<std::string, VecType>>, std::set<std::string>> GlobalIndexTable;
```

Comment:
`typedef` -> `using`
Also, is it worth converting `GlobalIndexTable` from `tuple<>` into a well-defined structure which explains all the details? It is possible that it might be extended in the future.